trace/google/cloud/trace/trace_context.py

```diff
@@ -0,0 +1,121 @@
+    :param trace_id: (Optional) Trace_id is a 32 digits uuid for the trace.
+                     If not given, will generate one automatically.
+
+    :type span_id: int
+    :param span_id: (Optional) Identifier for the span, unique within a trace.
+
+    :type enabled: bool
+    :param enabled: (Optional) Indicates whether the request is traced or not.
+    
+    :type from_header: bool
+    :param from_header: (Optional) Indicates whether the trace context is
+                        generated from request header.
+    """
+    def __init__(
+            self,
+            trace_id=None,
+            span_id=None,
+            enabled=True,
+            from_header=False):
+        if trace_id is None:
+            trace_id = generate_trace_id()
+
+        self.trace_id = trace_id
+        self.span_id = self.check_span_id(span_id)
+        self.enabled = enabled
+        self.from_header = from_header
+
+    def check_span_id(self, span_id):
+        """Check the type of span_id to ensure it is int. If it is not int,
+        first try to convert it to int, if failed to convert, then log a
+        warning message and set the span_id to None.
+
+        :type span_id: int
+        :param span_id: Identifier for the span, unique within a trace.
+
+        :rtype: int
+        :returns: Span_id for the current span.
+        """
+        if not isinstance(span_id, int):
+            try:
+                span_id = int(span_id)
+            except ValueError:
+                logging.warning(
+                    'The type of span_id should be int, got {}.'.format(
+                        span_id.__class__.__name__))
+                span_id = None
+
+        return span_id
+
+    def convert_to_string(self):
+        """Returns a string form of the TraceContext. This is the format of 
+        the Trace Context Header and should be forwarded to downstream
+        requests as the X-Cloud-Trace-Context header.
+
+        :rtype: str
+        :returns: String form of the TraceContext.
+        """
+        header = '{}/{};o={}'.format(
+            self.trace_id,
+            self.span_id,
+            self.enabled)
+        return header
+
+
+def generate_context_from_header(header):
+    """Parse a request header and generate a TraceContext object.
+
+    :type header: 
+    :param header: Request header from web application.
+
+    :rtype: :class:`~google.cloud.trace.trace_context.TraceContext`
+    :returns: A TraceContext object.
+    """
+    pattern = re.compile(_TRACE_CONTEXT_HEADER_FORMAT)
+    match = re.search(pattern, header)
+
+    if match:
+        trace_id = match.group(1)
+        span_id = match.group(3)
+        enabled = match.group(5)
+
+        if enabled is None:
+            enabled = 1
+
+        trace_context = TraceContext(
+            trace_id=trace_id,
+            span_id=span_id,
+            enabled=enabled,
+            from_header=True)
+        return trace_context
+    else:
+        return TraceContext()
```

Comment:
I think it might be possible that the web application is not running on GCP thus the header does not contain the trace_id, but it should still be able to trace the request.